Topics/Tech_Stacks/Introduction_to_Django_Templates.md

```diff
@@ -0,0 +1,193 @@
+# Introduction to Django Templates
+
+## Prerequisites
+- HTML &mdash; basic tags
+- CSS &mdash; selectors, colours
+- Django &mdash; project/app setup, URL design, function-based views
+
+## Introduction 
+Django projects typically process data in some way, such as tailoring information to a user. We may want to be able to serve these tailored webpages, but webpages developed with plain HTML do not support dynamic changes.
+
+Django Template Language (DTL) can be used to dynamically generate HTML pages, and other text-based formats such as XML and CSV. This article will introduce basic DTL syntax through generating a simple HTML webpage. As a running example, suppose we have the following static page for our WcDonald's Restaurant website:
+
+```
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>WcDonald's</title>
+    </head>
+    <body>
+      <h3>Menu</h3>
+          <ul>
+            <li>Chicken Wuggets $10.99</li>
+            <li>Big Wac $7.99</li>
+            <li>Morld Famous Fries $4.99</li>
+          </ul>
+    </body>
+</html>
+```
+
+What if we want to change the menu items? Or change the visibility of items on sale, which also change periodically? We can do so by creating a Django template. But first, we must introduce some setup and basic syntax.
+
+## Creating a Template
+First, assuming we have setup our Django project, create a folder called `templates` inside the app folder. This is where Django looks for templates by default. Inside, we create HTML templates by using the `.html` extension. The resulting folder structure may resemble the following: 
+
+```
+project/
+└── app/
+    └── templates/
+        └── index.html
+    └── ...
+```
+
+## Context
+Context is a dictionary of dynamic data that is passed to a template upon rendering. Later we will see a concrete example on how to do this in a function-based view.
+
+## Basic Syntax
+Django templates for webpages involve both static HTML tags, and DTL specific [syntax](https://docs.djangoproject.com/en/4.2/topics/templates/#syntax) including variables, tags, and filters.
+
+### Variables
+**Variable Substitution** is done by wrapping the variable name (passed in from the context) with double curly braces: 
+```
+<li>{{ menu_item }}</li>
+```
+So this would render as a list item whose text is the value of `menu_item`. If this variable doesn't exist in the given context, then its value is `None` by default, and it would simply render an empty string. However, this can be [configured](https://docs.djangoproject.com/en/1.11/ref/templates/api/#how-invalid-variables-are-handled).
+
+We can also pass in lists and dictionaries. Using a `.` performs a lookup to access by index/attribute:
+```
+<li>{{ menu_items.0.name }}</li>
+```
+
+Assuming menu_items is a list of dictionaries, this would display the first dictionary's name value.
```

Comment:
I think it would be nice to put "menu_items" in a code block to maintain consistency

Reply:
Same for "name" maybe?